slither/detectors/statements/wrong_encode_with_selector.py

```diff
@@ -0,0 +1,145 @@
+    """
+
+    ARGUMENT = "wrong-encode-selector"
+    HELP = "abi.encodeWithSelector with unexpected arguments"
+    IMPACT = DetectorClassification.MEDIUM
+    CONFIDENCE = DetectorClassification.HIGH
+
+    WIKI = "https://github.com/trailofbits/slither/wiki/encode-with-selector"
+    WIKI_TITLE = "Parameters of incorrect type or number in abi.encodeWithSelector"
+    WIKI_DESCRIPTION = "Wrong number or type of arguments in abi.encodeWithSelector"
+    WIKI_EXPLOIT_SCENARIO = """
+```solidity
+contract Test {   
+    event Val(uint, uint);
+    function f(uint a, uint b) public {
+        emit Val(a, b);
+    }
+}
+contract D {
+    function g() public {
+        Test t = new Test();
+        address(t).call(abi.encodeWithSelector(Test.f.selector, "test"));
+    }
+}
+```
+abi.encodeWithSelector's arguments do not match the types expected in the function signature.
+function signature.
+"""
+    WIKI_RECOMMENDATION = "Make sure that the type and number of arguments passed to abi.encodeWithSelector are the same as the target function signature."
+
+    def _detect(self) -> List[Output]:
+        # gather all known funcids
+        func_ids = {}
+        for contract in self.compilation_unit.contracts_derived:
+            func_ids.update(get_signatures(contract))
+        # todo: include func_ids from the public db
+
+        results = []
+        for contract in self.compilation_unit.contracts_derived:
+            for func, node in check_contract(contract, func_ids):
+                info = [
+                    func,
+                    " calls abi.encodeWithSelector() with arguments of incorrect type or with an incorrect number of arguments:",
+                    node,
+                ]
+                json = self.generate_result(info)
+                results.append(json)
+
+        return results
+
+
+def check_wrong_selector_encoding(
+    function: Function, node: Node, ir: Operation, func_ids: Dict[str, str]
+) -> List[Tuple[Function, Node]]:
+    result = []
+    if isinstance(ir, SolidityCall) and ir.function == SolidityFunction("abi.encodeWithSelector()"):
+        # build reference bindings dict
+        assignments = {}
+        for ir1 in node.irs:
+            if isinstance(ir1, Assignment):
+                assignments[ir1.lvalue.name] = ir1.rvalue
+
+        # if the selector is a reference, deref
+        selector = ir.arguments[0]
+        if isinstance(selector, ReferenceVariable):
+            selector = assignments[selector.name]
+
+        # Does not handle LocalVariables
+        if not isinstance(selector, Constant):
+            if isinstance(selector.expression, TypeConversion) and isinstance(
+                selector.expression.expression, Literal
+            ):
+                selector = selector.expression.expression.value
+            else:
+                return result
+
+        _, _, argument_types = func_ids[selector.value]
```

Comment:
This causes an error on 0x9078f5996746f134f761b93939b0fd595f68845e. We should return early if we cannot determine which function the selector map backs to until we integrate a selector db